docs/user-docs/build_grub.md

```diff
@@ -0,0 +1,221 @@
+mkdir build
+cd build
+../configure --with-platform=efi --target=x86_64
+
+# build
+make -j$(nproc)
+```
+
+### Setup
+
+```bash
+# also from inside build/ directory
+echo 'configfile ${cmdpath}/grub.cfg' > embedded.cfg
+./grub-mkimage -O x86_64-efi -o grubx64.efi -p / -c embedded.cfg -d grub-core \
+               all_video boot btrfs cat chain configfile echo efifwsetup \
+               efinet ext2 fat font gfxmenu gfxterm gzio halt hfsplus iso9660 \
+               jpeg loadenv loopback lvm mdraid09 mdraid1x minicmd normal \
+               part_apple part_msdos part_gpt password_pbkdf2 png reboot \
+               regexp search search_fs_uuid search_fs_file search_label serial \
+               sleep syslinuxcfg test tftp video xfs backtrace http linux usb \
+               usbserial_common usbserial_pl2303 usbserial_ftdi \
+               usbserial_usbdebug keylayouts at_keyboard multiboot2
+```
+
+The first command creates an embedded config which sources grub.cfg in the
+directory in which GRUB image is located.
+
+The second command produces the UEFI GRUB image named `grubx64.efi`.  The image
+along with a suitable `grub.cfg` needs to be installed on a EFI System
+Partition (ESP) and pointed at so firmware can find it (via Setup UI or a tool
+like `efibootmgr`).  It can also be started manually from a UEFI shell for a
+test.
+
+## Legacy version
+
+### Build
+
+```bash
+# check out the sources
+git clone --branch tb-2.12-57-v1 --depth 1 https://github.com/TrenchBoot/grub.git
+cd grub
+
+# make the checkout buildable
+./bootstrap
+
+# configure
+mkdir build
+cd build
+../configure --prefix=$PWD/local-install --target=i386
+
+# build
+make -j$(nproc)
+```
+
+### Setup
+
+Installation for legacy boot from inside of a container requires access to
+device files and mounting of boot directory.  For this reason the container
+needs to be started as `root` (inside of the container, don't prepend `sudo` to
+the command) and be a privileged one:
+
+```bash
+docker run --rm -it -v "$PWD:$PWD" -w "$PWD" \
+           --user root:root --privileged \
+           ghcr.io/trenchboot/trenchboot-sdk:master /bin/bash
+```
+
+```bash
+# if done in a container
+cd build
+mount BOOT-PARTITION PATH-TO-BOOT
+
+# also from inside build/ directory
+make install
+local-install/sbin/grub-install --target=i386-pc \
+                                --boot-directory PATH-TO-BOOT \
+                                DISK-DEVICE
+```
+
+In the second command above replace:
+
+- `BOOT-PARTITION` with the `/dev/...` file corresponding to the boot partition
+- `PATH-TO-BOOT` with the path to where the boot partition is mounted (`grub/`
+  directory will be created there)
+- `DISK-DEVICE` with the `/dev/...` file corresponding to the drive (not a
+  partition) on which GRUB is to be installed (double check before running the
+  command to avoid accidentally overwriting some working GRUB setup)
+
+If you get a weird error like `error: disk `hostdisk//dev/sda1' not found.`,
```

Comment:
```suggestion
If you get a weird error like ``error: disk `hostdisk//dev/sda1' not found.``,
```
`grub-install` outputs a back-tick there which breaks the rendering if not escaped

Reply:
https://github.com/TrenchBoot/TrenchBoot.github.io/compare/892edc3ec0dccbe1c9fc7564c5aa3f2f15f4f9e3..8df16c0628d723b775e5dd957f657159c8ab7b9a